src/tasks/tasks.py

```diff
@@ -2263,23 +2263,36 @@ def load_data(self):
 
         trg_map = {"true": 1, "false": 0, True: 1, False: 0}
 
+        def _process_sent_special(sent, word):
+            """ Tokenize the subsequence before the [first] instance of the word and after,
+            then concatenate everything together. This allows us to track where in the tokenized
+            sequence the marked word is located. """
+            sent_parts = sent.split(word)
+            sent_tok1 = process_sentence(self._tokenizer_name, sent_parts[0], self.max_seq_len)
+            sent_tok2 = process_sentence(self._tokenizer_name, sent_parts[1], self.max_seq_len)
+            sent_mid = process_sentence(self._tokenizer_name, word, self.max_seq_len)
+            sent_tok = sent_tok1[:-1] + sent_mid[1:-1] + sent_tok2[1:]
+            start_idx = len(sent_tok1[:-1]) - 1
+            end_idx = start_idx + len(sent_mid[1:-1])
+            return sent_tok, start_idx, end_idx
+
         def _load_split(data_file):
             sents1, sents2, idxs1, idxs2, trgs = [], [], [], [], []
             with open(data_file, "r") as data_fh:
                 for row in data_fh:
                     row = json.loads(row)
-                    sent1 = process_sentence(
-                        self._tokenizer_name, row["sentence1"], self.max_seq_len
+                    sent1 = row["sentence1"]
+                    sent2 = row["sentence2"]
+                    sent1, start1, end1 = _process_sent_special(
+                        sent1, sent1[row["start1"] : row["end1"]]
                     )
-                    sent2 = process_sentence(
-                        self._tokenizer_name, row["sentence2"], self.max_seq_len
+                    sent2, start2, end2 = _process_sent_special(
+                        sent2, sent2[row["start2"] : row["end2"]]
                     )
```

Comment:
Add an assert or test to make sure this does something reasonable (i.e., you're not deleting or doubling the target word).

Reply:
Hmm, any recommendations for how to do this? I'm not sure there's a straightforward way.

Reply:
Do some manual checking/verification for a couple of test tasks, and assert that (end_idx - start_idx) > 0.